builder/vsphere/driver/vm.go

```diff
@@ -499,13 +491,18 @@ func (vm *VirtualMachine) Configure(config *HardwareConfig) error {
 		confSpec.DeviceChange = append(confSpec.DeviceChange, spec)
 	}
 
-	if config.Firmware == "efi-secure" || config.Firmware == "efi" {
-		confSpec.Firmware = "efi"
-		confSpec.BootOptions = &types.VirtualMachineBootOptions{
-			EfiSecureBootEnabled: types.NewBool(config.Firmware == "efi-secure"),
-		}
-	} else if config.Firmware != "" {
-		confSpec.Firmware = config.Firmware
+	efiSecureBootEnabled := false
+	firmware := config.Firmware
```

Comment:
Looks like you removed all validation from the Firmware field. It may be nice to fail if the firmware value isn't valid.

Reply:
oh nevermind it's in the step_hardware Prepare already.